cpp/src/arrow/testing/random.cc

```diff
@@ -998,6 +1004,36 @@ std::shared_ptr<arrow::RecordBatch> GenerateBatch(const FieldVector& fields,
                                                   int64_t length, SeedType seed) {
   return RandomArrayGenerator(seed).BatchOf(fields, length);
 }
-
 }  // namespace random
+
+void rand_day_millis(int64_t N, std::vector<DayTimeIntervalType::DayMilliseconds>* out) {
+  const int random_seed = 0;
+  std::default_random_engine gen(random_seed);
```

Comment:
Since we're using `pcg32_fast` in RandomArrayGenerator, use it here too?

Reply:
done.